### ISSUE TITLE
MIT License?

### ISSUE THREAD

**Issue by xied75** [icsharpcode/SharpZipLib#103]
Dear Team,

Given the whole .NET world is now MIT, any chance this great lib could also change to that?

**Comment by christophwille** [comment]
This is currently also discussed in pull request https://github.com/icsharpcode/SharpZipLib/pull/101

**Comment by McNeight** [comment]
The following PRs were merged without MIT license assignment:
#1 by @linquize (2 lines in 1 file)
#4 by @Mailaender (2 lines in 1 file, and 2 deleted files)
#7 by @Mailaender (8 lines in one new file)
#10 by @RobThree (1 line in 1 file)
#53 by @novn2013 (1 line in 1 file)
#64 by @hypersw (10 lines in 1 file)
#65 by @ermshiperete and @hultqvist (16 lines in 6 files)
#66 by @benjamb (1 line in 1 file)

**Comment by hultqvist** [comment]
Do the listed PR contain enough to qualify for copyright and thus need the assignments. Looking at my own contribution (i changed the case in a few filenames) can't possibly reach verkshöjd(can't bother to translate) to be covered by copyright.

Although if you want me to take any legal action I will happily contact a lawyer and pass on the bill to you.

**Comment by McNeight** [comment]
Peter,

Inget behov av rättsliga åtgärder. Tack för att svara, och tack för ditt bidrag.

-Neil

**Comment by christophwille** [comment]
Good morning,

I am the "all things administrative red tape guy" on the SharpDevelop team. Some background on why we now want to (finally) take SharpZipLib to MIT - and action items.

@McNeight has joined as the new maintainer for SharpZipLib, and in the process of discussing with the old maintainers @jfreilly and @davidpierson somewhat obviously the license topic came up (and I can tell everybody that it was/is a pain to me too just by looking at the inbox on how many questions I got over the years just on license clarification).

Now, SharpZipLib started out of part of SharpDevelop, and with SharpDevelop we required JCAs (please see https://github.com/icsharpcode/SharpDevelop/wiki/Joining-the-Team). We want to get to exactly the same point with SharpZipLib, and so we want to go the following route:

 * The maintainer (@McNeight in this case) has a JCA on file. Check.
 * All PRs (that qualify as a patch) have the MIT/assignment paragraph from https://github.com/icsharpcode/SharpDevelop/wiki/Joining-the-Team as part of it. This issue is the thread for it.
 * The original port is cleaned up (which is necessary technically anyways)

About the PR part: usually in SharpDevelop we had the notion of "more than five lines" that needed a JCA. Which for the PRs @McNeight listed applies only to a few then, however, we would love that everyone adds the MIT paragraph to their past contributions (comment to the original PR). For documentation purposes and, basically, we want to be a good OSS community citizen as always.

Once that is done, we will have a contributing.md that reflects the "joing the team" document of SharpDevelop. Read: PRs come with that paragraph for documentation purposes (we want to do away with the paperwork (even digital)).

@Mailaender, @hypersw, @ermshiperete and @hultqvist: could you please amend your past PRs with 

I certify that I own, and have sufficient rights to contribute, all source code and related material intended to be compiled or integrated with the source code for the SharpZipLib open source product (the "Contribution"). My Contribution is licensed under the MIT License.

All others mentioned by @McNeight - not strictly required but we would love you to do that too.

Thank you in helping us make the license of SharpZipLib easy & palatable to an even broader audience!

**Comment by Mailaender** [comment]
Yeah sure, although I guess my contributions are somehow insignificant.

**Comment by McNeight** [comment]
I would like to merge the following PRs once we have an MIT license assignment from the original author:
#18 by @hempels (5 lines in 1 file)
#69 by @Ezghoul (2 lines in 1 file)
#74 by @konstantingolovanov (26 lines in 1 file)
#76 by @bbenoist (2 lines in 1 file)
#83 by @depend86 (64 lines in 1 file)
#84 by @kkguo (2 lines in 1 file)
#85 by @pldg (5 lines in 1 file)
#86 by @creker (7 lines in 2 files)
#93 by @bastianeicher (102 lines in 6 files)
#96 by @aviranco (34 lines in 1 file)

**Comment by konstantingolovanov** [comment]
Yes sure.

**Comment by konstantingolovanov** [comment]
Yes, certainly.

15.04.2016, 01:56, "Neil McNeight" <notifications@github.com>:

**Comment by pldg** [comment]
I certify that Google owns, and I have sufficient rights to contribute, all
source code and related material for pull request 85 by @pldg intended to
be compiled or integrated with the source code for the #develop open source
product (the "Contribution"). My Contribution is licensed under the MIT
License.

**Comment by konstantingolovanov** [comment]
I certify that I own, and have sufficient rights to contribute, all source code and related material for pull request #74 by @konstantingolovanov intended to be compiled or integrated with the source code for the SharpZipLib open source product (the "Contribution"). My Contribution is licensed under the MIT License.

**Comment by hempels** [comment]
I certify that Renaissance Learning Inc owns, and I have sufficient rights to contribute, all source code and related material for pull request #18 by @hempels intended to be compiled or integrated with the source code for the SharpZipLib open source product (the "Contribution"). My Contribution is licensed under the MIT License.

**Comment by kkguo** [comment]
I certify that I own, and have sufficient rights to contribute, all source code and related material for pull request #84 by @kkguo intended to be compiled or integrated with the source code for the SharpZipLib open source product (the "Contribution"). My Contribution is licensed under the MIT License.

**Comment by Ezghoul** [comment]
I certify that I own, and have sufficient rights to contribute, all source code and related material for pull request #69 by @Ezghoul intended to be compiled or integrated with the source code for the SharpZipLib open source product (the "Contribution"). My Contribution is licensed under the MIT License.

**Comment by ReinoutW** [comment]
Shouldn't the change in license also be reflected on http://icsharpcode.github.io/SharpZipLib/ ?
It still reads:

> License
> The library is released under the GPL with the following exception:
> 
> Linking this library statically or dynamically with other modules is making a combined work based on this library. Thus, the terms and conditions of the GNU General Public License cover the whole combination.
> 
> As a special exception, the copyright holders of this library give you permission to link this library with independent modules to produce an executable, regardless of the license terms of these independent modules, and to copy and distribute the resulting executable under terms of your choice, provided that you also meet, for each linked independent module, the terms and conditions of the license of that module. An independent module is a module which is not derived from or based on this library. If you modify this library, you may extend this exception to your version of the library, but you are not obligated to do so. If you do not wish to do so, delete this exception statement from your version.
> 
> Note The exception is changed to reflect the latest GNU Classpath exception. Older versions of #ziplib did have another exception, but the new one is clearer and it doesn't break compatibility with the old one.
> 
> Bottom line In plain English this means you can use this library in commercial closed-source applications.

It is somewhat confusing to see different info on https://github.com/icsharpcode/SharpZipLib

> License
> 
> This software is now released under the MIT License. Please see issue #103 for more information on the relicensing effort.
> 
> Previous versions were released under the GNU General Public License, version 2 with an exception which allowed linking with non-GPL programs.

**Comment by ntjtomohide** [comment]
I think so.I want to use this library, but it's license policy confused. So I cannot use this library on MIT License.

**Comment by SOesterreicher** [comment]
Hello, any news on the license question in regards of the License Change on the http://icsharpcode.github.io/SharpZipLib/ page to the github provided license? 
Thanks in advance.

**Comment by brendonferris** [comment]
Hi - This looks like a great library and I would like to use it.  

However, as @SOesterreicher mentioned,  http://icsharpcode.github.io/SharpZipLib/ is still showing that the licence as GPL.

1) Is it now possible to update http://icsharpcode.github.io/SharpZipLib/ to show that it is MIT?

2) Can you confirm that it is definitely MIT now (and there are no outstanding licensing issues)?

Thank you.

**Comment by konstantingolovanov** [comment]
Good day.I confirm that it is definitly MIT license.Thanks.--Отправлено из мобильной Яндекс.Почты23.10.2018, 05:37, "brendonferris" <notifications@github.com>:Hi - This looks like a great library and I would like to use it.
However, as @SOesterreicher mentioned,  http://icsharpcode.github.io/SharpZipLib/ is still showing that the licence as GPL.


Is it now possible to update http://icsharpcode.github.io/SharpZipLib/ to show that it is MIT?


Can you confirm that it is definitely MIT now (and there are no outstanding licensing issues)?


Thank you.

**Comment by git-hub-9999** [comment]
Great - thank you for the confirmation.

**Comment by lytico** [comment]
I certify that I own, and have sufficient rights to contribute, all source code and related material for
https://github.com/icsharpcode/SharpZipLib/blob/8feb39a0c26e83b80e7fb0c481834e88d35297ee/docs/Changes.txt @lytico intended to be compiled or integrated with the source code for the SharpZipLib open source product (the "Contribution"). My Contribution is licensed under the MIT License.

**Comment by konstantingolovanov** [comment]
MIT License forever--Sent from Yandex.Mail for mobile30.12.2019, 23:39, "nils måsén" <notifications@github.com>:Closed #103.
